### PR TITLE
Fix 703: correct CrsMatrix documentation

### PR DESCRIPTION
**Pull request by brian-kelley**
for raw host array constructor. It said it takes COO format, but it really takes CRS format. Also rename "rows" to "rowmap" to avoid confusion. Also remove unused parameter "pad". I'm 99% sure nobody ever passed anything in for pad, but I'm running spot checks just in case.

### CODE REVIEW

File: src/sparse/KokkosSparse_CrsMatrix.hpp [hunk 1]
```diff
@@ -519,27 +519,21 @@ class CrsMatrix {
...
   CrsMatrix (const std::string &label,
              OrdinalType nrows,
              OrdinalType ncols,
              size_type annz,
              ScalarType* val,
-             OrdinalType* rows,
-             OrdinalType* cols,
-             bool pad = false)
+             OrdinalType* rowmap,
+             OrdinalType* cols)
```
Comment (srajama1) [inline src/sparse/KokkosSparse_CrsMatrix.hpp:534]: You are sure Tpetra doesn't call this constructor w/ pads set ?

Comment (brian-kelley) [reply src/sparse/KokkosSparse_CrsMatrix.hpp:534]: @srajama1 I'll find out. But the pad argument never had any effect, so I doubt it. We can also catch and fix that easily during promotion if anyone did pass in something.

**Comment by brian-kelley** [comment]
@srajama1 I did an all-package Trilinos build with the parameter removed and it was OK, so I think this is ready.

**Comment by brian-kelley** [comment]
Oops, MueLu did pass in pad in one place. I'll get rid of it and explain that the parameter has no  effect and a default value.

**Comment by srajama1** [comment]
Yeah, that is what I was worried about. We need to fix MueLU when this gets into Trilinos then. Technically, this is a backwards incompatible change, so we shouldn't do it. However, this is such an obscure future we might be ok.

**Comment by brian-kelley** [comment]
@srajama1 Just opened https://github.com/trilinos/Trilinos/pull/7280 to remove it in MueLu, so that this won't come up when we put together the next snapshot.

**Comment by srajama1** [comment]
Thanks, @brian-kelley !

**Pull request merged**